onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -999,7 +1012,24 @@ struct DFTShapeInferrer : public WRL::Base<IMLOperatorShapeInferrer>
                 throw;
             }
 
-            auto axisIdx = OperatorHelper::HandleNegativeAxis(static_cast<int32_t>(axis), rank);
+            int64_t axisValue = -2;
+            bool isDft17 = !context->IsInputValid(2);
+
+            if (isDft17)
```

Comment:
in the method that you modified above, you changed the signature to take the version as an input, and here you're inferring it from the context. Is there a reason they're different? Are they equivalent? Does inferring from the context still work for future versions?

Reply:
conversely - does it work to make the method not care about the version at all? Specifically, check for an "axis" attribute (like below) and if it's there, use it. Else, check for an axis tensor, and use it if it's there. And then (perhaps) rely on the onnx schema validation to guarantee that the version matches with the expected attributes and inputs?

Reply:
They are equivalent; the DFTShapeInferrer struct uses the context check as extra validation to ensure the inputs match. I think it makes sense to use the version as the main decider and then check that the attributes and inputs match rather than the reverse since the version is specified during registration.